docker/ci/dockerfile

```diff
@@ -35,11 +36,21 @@ RUN apt update -qq \
     && update-alternatives --install /usr/bin/gcov-tool gcov-tool /usr/bin/gcov-tool-${GCC_VERSION} 100 \
     && update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-${LLVM_TOOLS_VERSION} 100 \
     && apt-get clean && apt remove -y software-properties-common \
-    && pip3 install  -q --upgrade --no-cache-dir pip \
-    && pip3 install  -q --no-cache-dir conan==1.62 gcovr cmake-format
+    && pip3 install -q --upgrade --no-cache-dir pip \
+    && pip3 install -q --no-cache-dir conan==1.62 gcovr cmake-format
 
 WORKDIR /tmp
 
+# Install gcc-12 from source
+RUN wget "http://mirrors.concertpass.com/gcc/releases/gcc-${GCC12_VERSION}/gcc-${GCC12_VERSION}.tar.gz" \
+    && tar xf "gcc-${GCC12_VERSION}.tar.gz" \
+    && cd "gcc-${GCC12_VERSION}" \
+    && mkdir build && cd build \
+    && apt-get install -y libmpc-dev libgmp3-dev libmpfr-dev \
+    && ../configure --enable-languages=c,c++ --disable-multilib \
+    && make -j$(($(nproc --all) * 2)) BOOT_CFLAGS='-O3 -g0' profiledbootstrap \
```

Comment:
Not too sure about this `-j` setting. It works on my machine but it may kill our runner. Perhaps this is too aggressive and we need to make it somewhat smaller.

Reply:
I think `-j$(nproc)` would be fine.